diff --git a/drivers/pci/msi.c b/drivers/pci/msi.c
index 0099a00af361..620928fd0065 100644
--- a/drivers/pci/msi.c
+++ b/drivers/pci/msi.c
@@ -148,6 +148,9 @@ static noinline void pci_msi_update_mask(struct msi_desc *desc, u32 clear, u32 s
 	raw_spinlock_t *lock = &desc->dev->msi_lock;
 	unsigned long flags;
 
+	if (pci_msi_ignore_mask)
+		return;
+
 	raw_spin_lock_irqsave(lock, flags);
 	desc->msi_mask &= ~clear;
 	desc->msi_mask |= set;
@@ -179,6 +182,9 @@ static inline void __iomem *pci_msix_desc_addr(struct msi_desc *desc)
  */
 static void pci_msix_write_vector_ctrl(struct msi_desc *desc, u32 ctrl)
 {
+	if (pci_msi_ignore_mask)
+		return;
+
 	void __iomem *desc_addr = pci_msix_desc_addr(desc);
 
 	writel(ctrl, desc_addr + PCI_MSIX_ENTRY_VECTOR_CTRL);
@@ -186,6 +192,9 @@ static void pci_msix_write_vector_ctrl(struct msi_desc *desc, u32 ctrl)
 
 static inline void pci_msix_mask(struct msi_desc *desc)
 {
+	if (pci_msi_ignore_mask)
+		return;
+
 	desc->msix_ctrl |= PCI_MSIX_ENTRY_CTRL_MASKBIT;
 	pci_msix_write_vector_ctrl(desc, desc->msix_ctrl);
 	/* Flush write to device */
@@ -194,6 +203,9 @@ static inline void pci_msix_mask(struct msi_desc *desc)
 
 static inline void pci_msix_unmask(struct msi_desc *desc)
 {
+	if (pci_msi_ignore_mask)
+		return;
+
 	desc->msix_ctrl &= ~PCI_MSIX_ENTRY_CTRL_MASKBIT;
 	pci_msix_write_vector_ctrl(desc, desc->msix_ctrl);
 }
-- 
2.26.2

